server/server_test.go

```diff
@@ -3219,6 +3229,78 @@ func newClient(t *testing.T, pool *x509.CertPool, clientKeyPair ...string) *http
 	return &c
 }
 
+func TestH2CHTTPListeners(t *testing.T) {
```

Comment:
The majority of other tests in here, afaik, don't actually start the loops. They are mostly testing the HTTP handler directly by calling `Handle()` on it with a recorder and some request. I'd recommend moving this test into a new package in https://github.com/open-policy-agent/opa/tree/master/test/e2e where we have more tests that are like this one which are testing out the sort of full extent of the API service (plus you get a bunch of helpers over there for stuff like cleaning up, waiting for the server, getting the addresses, etc).

Reply:
Awesome! I'll move it. 😃

Reply:
Done. It's nicer indeed. I've got a test or two to move there for authentication with TLS and authorization, I think, but that'll be a follow-up.